src/main/java/com/cleo/rest/services/impl/CarsWebService.java

```diff
@@ -1,30 +1,333 @@
+   * filter and sort.
+   *
+   *
+   * @param filter -> List cars by a specific attribute (id, make, model, color, year).
+   * @param sort -> Specify order of cars based on filter to be in ascending (asc) or descending (desc) order.
+   * @return -> Display of cars based on user's query parameters.
+   */
+  @Override
+  public Response getAll(String filter, String sort) {
+
+    if(filter.equals("id")) {
+      Comparator<Car> byId = Comparator.comparing(Car::getId);
+      if(sort.equals("asc")) {
+        return Response.ok(
+            cars.stream()
+                .sorted(byId)
+                .collect(Collectors.toList())
+        ).build();
+      }
+      else if(sort.equals("desc")) {
+        return Response.ok(
+            cars.stream()
+                .sorted(byId.reversed())
+                .collect(Collectors.toList())
+        ).build();
+      }
+    }
+
+    if(filter.equals("make")) {
+      Comparator<Car> byMake = Comparator.comparing(Car::getMake);
+      if(sort.equals("asc")) {
+        return Response.ok(
+            cars.stream()
+                .sorted(byMake)
+                .collect(Collectors.toList())
+        ).build();
+      }
+      else if(sort.equals("desc")) {
+        return Response.ok(
+            cars.stream()
+                .sorted(byMake.reversed())
+                .collect(Collectors.toList())
+        ).build();
+      }
+    }
+
+    if(filter.equals("model")) {
+      Comparator<Car> byModel = Comparator.comparing(Car::getModel);
+      if(sort.equals("asc")) {
+        return Response.ok(
+            cars.stream()
+                .sorted(byModel)
+                .collect(Collectors.toList())
+        ).build();
+      }
+      else if(sort.equals("desc")) {
+        return Response.ok(
+            cars.stream()
+                .sorted(byModel.reversed())
+                .collect(Collectors.toList())
+        ).build();
+      }
+    }
+
+    if(filter.equals("color")) {
+      Comparator<Car> byColor = Comparator.comparing(Car::getColor);
+      if(sort.equals("asc")) {
+        return Response.ok(
+            cars.stream()
+                .sorted(byColor)
+                .collect(Collectors.toList())
+        ).build();
+      }
+      else if(sort.equals("desc")) {
+        return Response.ok(
+            cars.stream()
+                .sorted(byColor.reversed())
+                .collect(Collectors.toList())
+        ).build();
+      }
+    }
+
+    if(filter.equals("year")) {
+      Comparator<Car> byYear = Comparator.comparing(Car::getYear);
+      if(sort.equals("asc")) {
+        return Response.ok(
+            cars.stream()
+                .sorted(byYear)
+                .collect(Collectors.toList())
+        ).build();
+      }
+      else if(sort.equals("desc")) {
+        return Response.ok(
+            cars.stream()
+                .sorted(byYear.reversed())
+                .collect(Collectors.toList())
+        ).build();
+
+      }
+    }
```

Comment:
Looks like we have a decent amount of duplication here. It's probably out of the scope of this exercise, but try to think of some alternate ways to do this in order to remove some of the duplication. I would recommend something like a `Map<String, Comparator<Car>>` where the key is the `filter`.

Then you could just have one call that looks something like:
```java
cars.stream()
    .sorted(SORTERS.get(filter))
    .collect(Collectors.toList());
```

Btw, nice find on the `@DefaultValue` annotation. Any time we can cut out lines of code with a simple annotation like that it's usually a good thing.